eng/SourceBuildPrebuiltBaseline.xml

```diff
@@ -12,6 +12,9 @@
     <UsagePattern IdentityGlob="System.Reflection.MetadataLoadContext/*7.0.0*" />
     <UsagePattern IdentityGlob="System.Security.Cryptography.ProtectedData/*7.0.0*" />
     <UsagePattern IdentityGlob="System.Text.Json/*7.0.3*" />
+
+    <!-- This is for windows-only project caching acceleration -->
+    <UsagePattern IdentityGlob="System.Security.Principal.Windows/*6.0.0*" />
```

Comment:
This is a prebuilt and adding an exclusion here is going to break the product source-build.  Previously it looks like version 5.0.0 was referenced.  A [reference package](https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/system.security.principal.windows/5.0.0) was being utilized during source-build to satisfy the windows-only dependency.  Since this is being upgraded, a new reference package will need to be created.  You can add this by following the instructions at https://github.com/dotnet/source-build-reference-packages/blob/main/README.md#adding-new-packages.

One question I have is why are we taking a dependency on a preview package?  We frown upon adding reference packages for non-stable versions.  I see this package has been in preview for over 2 years.  That smells to me.

Reply:
Looks like this is moot and I can revert the change to this package version. BXL's package only depends on this version for net6, and since originally sending this PR MSBuild has updated to net7.

![image](https://github.com/dotnet/msbuild/assets/6445614/486504d1-09e2-4087-93b5-9fa120928a5f)